nat-lab/tests/test_mesh_connected_socket.py

```diff
@@ -0,0 +1,105 @@
+import asyncio
+import pytest
+import telio
+from contextlib import AsyncExitStack
+from helpers import SetupParameters, setup_mesh_nodes
+from telio import PathType, State
+from telio_features import TelioFeatures, Direct
+from utils.connection_util import ConnectionTag
+from utils.iperf3 import IperfServer, IperfClient, Protocol
+from utils.ping import Ping
+
+
+@pytest.mark.asyncio
+@pytest.mark.long
+@pytest.mark.parametrize(
+    "disable_connected_socket_a",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "disable_connected_socket_b",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "beta_tag",
+    [ConnectionTag.MAC_VM, ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_2],
+)
+async def test_mesh_connected_socket(
+    disable_connected_socket_a, disable_connected_socket_b, beta_tag
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        features_alpha = TelioFeatures(
+            direct=Direct(providers=["stun", "local"]),
+            disable_connected_socket=disable_connected_socket_a,
+        )
+        features_beta = TelioFeatures(
+            direct=Direct(providers=["stun", "local"]),
+            disable_connected_socket=disable_connected_socket_b,
+        )
+        env = await setup_mesh_nodes(
+            exit_stack,
+            [
+                SetupParameters(
+                    connection_tag=ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_1,
+                    adapter_type=telio.AdapterType.BoringTun,
+                    features=features_alpha,
+                ),
+                SetupParameters(
+                    connection_tag=beta_tag,
+                    adapter_type=telio.AdapterType.BoringTun,
+                    features=features_beta,
+                ),
+            ],
+        )
+        alpha, beta = env.nodes
+
+        client_alpha, client_beta = env.clients
+        connection_alpha, connection_beta = [
+            conn.connection for conn in env.connections
+        ]
+
+        asyncio.gather(
+            client_alpha.wait_for_state_peer(
+                beta.public_key, [State.Connected], [PathType.Direct]
+            ),
+            client_beta.wait_for_state_peer(
+                alpha.public_key, [State.Connected], [PathType.Direct]
+            ),
+        )
```

Comment:
This is already done by `setup_mesh_nodes` since direct connections are enabled

Reply:
Good point, I'll simplify it, thanks!